doc/user_guide/user_guide.md

```diff
@@ -0,0 +1,291 @@
+| `<TEMP_DB_SCHEMA>`           | yes       | Database Schema for temporary database objects, e.g. tables                                                    |
+
+# Implementation of Custom Algorithms
+
+Each algorithm should extend the `UDFQueryHandler` abstract class and then implement the following methods:
+* `start()`: This method is called at the first execution of the framework, that is, in the first iteration. It returns a result object: Either _Finish_ or _Continue_.
+  * The _Finish_ result object contains the final result of the implemented algorithm.
+  * The _Continue_ object contains the query list that will be executed for the next state.
+* `handle_query_result()`: This method is called at the following iterations to handle the return query.
+
+Here is an example class definition:
+
+```python
+--/
+CREATE OR REPLACE PYTHON3_AAF SET SCRIPT "MY_SCHEMA"."MY_QUERY_HANDLER_UDF"(...)
+EMITS (outputs VARCHAR(2000000)) AS
+
+from typing import Union
+from exasol_advanced_analytics_framework.udf_framework.udf_query_handler import UDFQueryHandler
+from exasol_advanced_analytics_framework.query_handler.context.query_handler_context import QueryHandlerContext
+from exasol_advanced_analytics_framework.query_result.query_result import QueryResult
+from exasol_advanced_analytics_framework.query_handler.result import Result, Continue, Finish
+from exasol_advanced_analytics_framework.query_handler.query.select_query import SelectQuery, SelectQueryWithColumnDefinition
+from exasol_data_science_utils_python.schema.column import Column
+from exasol_data_science_utils_python.schema.column_name import ColumnName
+from exasol_data_science_utils_python.schema.column_type import ColumnType
+
+
+-- proposal write parameter into temp table in schema ...
+class CustomQueryHandler(UDFQueryHandler):
+    def __init__(self, parameter: str, query_handler_context: QueryHandlerContext):
+        super().__init__(parameter, query_handler_context)
+        self.parameter = parameter
+        self.query_handler_context = query_handler_context
+
+    def start(self) -> Union[Continue, Finish[str]]:
+        query_list = [
+          SelectQuery("SELECT 1 FROM DUAL"),
+          SelectQuery("SELECT 2 FROM DUAL")]
+        query_handler_return_query = SelectQueryWithColumnDefinition(
+            query_string="SELECT 5 AS 'return_column' FROM DUAL",
+            output_columns=[
+              Column(ColumnName("return_column"), ColumnType("INTEGER"))])
+
+        return Continue(
+            query_list=query_list,
+            input_query=query_handler_return_query)
+
+    def handle_query_result(self, query_result: QueryResult) -> Union[Continue, Finish[str]]:
+        return_value = query_result.return_column
+        result = 2 ** return_value
+        return Finish(result=f"Assertion of the final result: 32 == {result}")
+
+import builtins
+builtins.CustomQueryHandler=CustomQueryHandler # required for pickle
```

Comment:
in a extra PR we need to add a function like register_query_handler_inside_udf and comment that this is only needed when you define the query handler insidde of a UDF